TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -239,6 +239,83 @@ public override bool Visit(Move move)
             return true;
         }
 
+        public override bool Visit(WhenSearch whenSearch)
+        {
+            var search = whenSearch.Parent;
+            System.Diagnostics.Debug.Assert(search is Search);
+            if (search.CodeElement is SearchBinaryStatement)
```

Comment:
Test on `CodeElement.Type` is more efficient. Sorry for the late comment, should have picked up in my previous review.

Reply:
In that case we specifically want `SearchBinaryStatement` (SEARCH ALL) and not `SearchSerialStatement` (SEARCH VARYING).
The distinction is not available in `CodeElementType`.

Reply:
You're right, the info is stored in `StatementType` property of `StatementElement` but it would still require a type-check-cast to access it so no need to change that.